lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -173,8 +173,8 @@ def initialization_script
       su_method = @user_method.instance_of?(ForemanRemoteExecutionCore::SuUserMethod)
       # pipe the output to tee while capturing the exit code in a file
       <<-SCRIPT.gsub(/^\s+\| /, '')
-      | sh -c "(#{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}; echo \\$?>#{@exit_code_path}) | /usr/bin/tee #{@output_path}
-      | exit \\$(cat #{@exit_code_path})"
+      | sh -c "(#{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script.shellescape.gsub(/"/,'\\\\\\"')} < /dev/null"}; echo \\$?>#{@exit_code_path.shellescape.gsub(/"/,'\\\\\\"')}) | /usr/bin/tee #{@output_path.shellescape.gsub(/"/,'\\\\\\"')}
+      | exit \\$(cat #{@exit_code_path.shellescape.gsub(/"/,'\\\\\\"')})"
```

Comment:
There is a way how we could avoid this substitution and few of the escapes we currently do by hand

```shell
# Instead of current
sh -c "($content) | tee $output"

# we could use
sh <<EOF | tee $output
$content
EOF
```